### PR TITLE
move pybind11 to python3 only 

### PR DESCRIPTION
**Pull request by davidlange6**
This PR takes advantage of pybind11 being used as a header only package having only a loose connection to python version. The pybind11 toolfile should be renamed. This will be done as a follow-on PR as it also involves cmssw changes

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by mrodozov** [comment]
the test is failing, I think it's scram (or something scram doesn't read as expected)
```
13:09:38 Name : py2-pybind11
13:09:38 Traceback (most recent call last):
13:09:38   File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/share/lcg/SCRAMV1/V3_00_23/bin/scram.py", line 114, in <module>
13:09:38     main()
13:09:38   File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/share/lcg/SCRAMV1/V3_00_23/bin/scram.py", line 109, in main
13:09:38     if not execcommand(args, opts):
13:09:38   File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/share/lcg/SCRAMV1/V3_00_23/bin/scram.py", line 103, in execcommand
13:09:38     return eval('scram_commands.cmd_%s' % cmds[0])(args, opts)
13:09:38   File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/share/lcg/SCRAMV1/V3_00_23/SCRAM/Core/CMD.py", line 43, in cmd_setup
13:09:38     return process(args)
13:09:38   File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/share/lcg/SCRAMV1/V3_00_23/SCRAM/Core/Commands/setup.py", line 60, in process
13:09:38     toolmanager.setupalltools()
13:09:38   File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/share/lcg/SCRAMV1/V3_00_23/SCRAM/BuildSystem/ToolManager.py", line 64, in setupalltools
13:09:38     self.coresetup(toolfile, dump)
13:09:38   File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/share/lcg/SCRAMV1/V3_00_23/SCRAM/BuildSystem/ToolManager.py", line 86, in coresetup
13:09:38     self._update_json(tooljson, dump)
13:09:38   File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/share/lcg/SCRAMV1/V3_00_23/SCRAM/BuildSystem/ToolManager.py", line 113, in _update_json
13:09:38     printmsg("Version : %s" % self.xml.contents['TOOLVERSION'])
13:09:38 KeyError: 'TOOLVERSION'
13:09:38 + ERR=1
```

**Comment by davidlange6** [comment]
Thanks for the heads up - I guess I should build the entire stack to debug this  - I realize that I probably needed to add a python2 requirement to py2-pybind11-toolfile.

File: py2-pybind11-toolfile.spec [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ### RPM external py2-pybind11-toolfile 1.0
```
Comment (smuzaffar) [inline py2-pybind11-toolfile.spec]: @davidlange6 , rename the spec to `py3-pybind11.spec`  and also update the name in the line above

**Comment by smuzaffar** [comment]
@davidlange6 , as cmsRun still uses py2, so we will still need the py2-pybind11

**Comment by davidlange6** [comment]
I don’t think so - Pybind11 is just headers.. notice that we use py2-pybind11 for both the py2 and py3 cmsRuns currently (an oversight..)

**Comment by davidlange6** [comment]
please test

(I'm building cmssw locally as well - but it seems better now)

**Comment by smuzaffar** [comment]
@davidlange6 , did you see my comment https://github.com/cms-sw/cmsdist/pull/6996/files#r647386284 ? the toolfile name and package name should match. if package is called py3-pybind11 then toolfile spec should be py3-pybind11-toolfile  otherwise the logic https://github.com/cms-sw/cmsdist/blob/IB/CMSSW_12_0_X/master/scram-tools-post.file#L4-L7 fails

**Comment by davidlange6** [comment]
ok, no - I didn't. - I'll abort and find another approach.

**Comment by davidlange6** [comment]
please abort

**Comment by davidlange6** [comment]
this will be tested with cmssw changes in #34026..

File: py3-pybind11-toolfile.spec [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Requires: py3-pybind11
+
+%prep
+
+%build
+
+%install
+
+mkdir -p %{i}/etc/scram.d
+cat << \EOF_TOOLFILE >%{i}/etc/scram.d/py2-pybind11.xml
```
Comment (smuzaffar) [inline py3-pybind11-toolfile.spec]: @davidlange6 , please also update the xml file name. It should be py3-pybind11.xml

**Comment by davidlange6** [comment]
Argh - I missed that -sorry

**Comment by smuzaffar** [comment]
please test

**Comment by davidlange6** [comment]
Will fail if its not tested with the cmssw changes.. (I think I started those tests)

**Comment by qliphy** [comment]
tested with https://github.com/cms-sw/cmssw/pull/34026

**Comment by qliphy** [comment]
+1

**Comment by qliphy** [comment]
merge

**Comment by ggovi** [comment]
+1

**Pull request merged**